[typing/runtime] [gql] eliminate resolver **kwargs (3)

smackesey opened the pull request:
### Summary & Motivation

Replace resolver `**kwargs` with typed keyword args. Part of a collection of PRs that eliminates resolver `**kwargs` across all resolvers.

**NOTE**: The GQL schema in this PR is changed, but its a non-functional change-- the change arises from the parameter order of the signature of `resolve_isPipelineConfigValid`-- keyword args with defaults need to appear at the end of the param list in a python function. This changes the order in which the args are listed in the generated schema, but this doesn't matter because args are always passed by name in GQL.

### How I Tested These Changes

BK. Also, correctness of added args was confirmed by manually inspecting the field definition in the parent object.

smackesey commented:
Current dependencies on/for this PR:
* master
  * **PR #12242** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12242" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12242?utm_source=stack-comment).

alangenfeld commented on python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py:
```diff
@@ -596,43 +629,69 @@ def resolve_pipelineRunTags(self, graphene_info: ResolveInfo):
+        graphene_info: ResolveInfo,
+        pipeline: GraphenePipelineSelector,
+        mode: str,
+        runConfigData: Optional[GrapheneRunConfigData] = None,
```
[1] since GrapheneRunConfigData represents a "custom scalar", i don't think using that type here is right. Theoretically we could implement more on that class and align to clearer types that way. For now might be best to just do Any / Optional[object] / what we expect which is what `parse_run_config_input` is typed to.

https://docs.graphene-python.org/en/latest/types/scalars/#custom-scalars

alangenfeld commented on python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py:
```diff
@@ -596,43 +629,69 @@ def resolve_pipelineRunTags(self, graphene_info: ResolveInfo):
+        graphene_info: ResolveInfo,
+        pipeline: GraphenePipelineSelector,
+        mode: str,
+        runConfigData: Optional[GrapheneRunConfigData] = None,
```
[1]

alangenfeld commented on python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py:
```diff
@@ -596,43 +629,69 @@ def resolve_pipelineRunTags(self, graphene_info: ResolveInfo):
+        assetKeys: Optional[Sequence[GrapheneAssetKeyInput]] = None,
+    ) -> Sequence[GrapheneAssetNode]:
+        resolved_asset_keys = set(
+            AssetKey.from_graphql_input(cast(Mapping[str, Sequence[str]], asset_key_input))
```
[2] this method exists for these call site, so it feels off that it requires casting like this. Seems likely worth updating the method definition.

smackesey replied:
Good catch, it actually doesn't require a `cast`, this was left over. `GrapheneAssetKeyInput`, like all Graphene input objects, functions as an implicit `Any`.

alangenfeld commented on python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py:
```diff
@@ -708,45 +770,62 @@ def load_asset_graph() -> ExternalAssetGraph:
-    def resolve_assetNodeOrError(self, graphene_info: ResolveInfo, **kwargs):
-        asset_key_input = cast(Mapping[str, Sequence[str]], kwargs.get("assetKey"))
+    def resolve_assetNodeOrError(self, graphene_info: ResolveInfo, assetKey: GrapheneAssetKeyInput):
+        asset_key_input = cast(Mapping[str, Sequence[str]], assetKey)
```
[2]

alangenfeld commented on python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py:
```diff
@@ -708,45 +770,62 @@ def load_asset_graph() -> ExternalAssetGraph:
-    def resolve_assetOrError(self, graphene_info: ResolveInfo, **kwargs):
-        return get_asset(graphene_info, AssetKey.from_graphql_input(kwargs["assetKey"]))
+    def resolve_assetOrError(self, graphene_info: ResolveInfo, assetKey: GrapheneAssetKeyInput):
+        return get_asset(graphene_info, AssetKey.from_graphql_input(assetKey))
```
[2] how is this one fine but the others needed casting?

alangenfeld commented on python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py:
```diff
@@ -708,45 +770,62 @@ def load_asset_graph() -> ExternalAssetGraph:
+        assetKeys: Optional[Sequence[GrapheneAssetKeyInput]] = None,
+    ):
+        assert assetKeys is not None
+        raw_asset_keys = cast(Sequence[Mapping[str, Sequence[str]]], assetKeys)
```
[2]

smackesey commented:
@alangenfeld Including fix for BK / addressing your comments on https://github.com/dagster-io/dagster/pull/12244 here since I accidentally merged that one before comment address.
